api/controllers/linux/install/statemachine.go

```diff
@@ -24,9 +24,16 @@ var validStateTransitions = map[statemachine.State][]statemachine.State{
 	states.StateHostPreflightsSucceeded:         {states.StateApplicationConfiguring, states.StateInstallationConfiguring, states.StateHostConfiguring, states.StateHostPreflightsRunning, states.StateInfrastructureInstalling},
 	states.StateHostPreflightsFailedBypassed:    {states.StateApplicationConfiguring, states.StateInstallationConfiguring, states.StateHostConfiguring, states.StateHostPreflightsRunning, states.StateInfrastructureInstalling},
 	states.StateInfrastructureInstalling:        {states.StateSucceeded, states.StateInfrastructureInstallFailed},
+	// TODO: only allow running preflights after infra is installed and before installing the app once the app installation is decoupled from infra installation
+	states.StateAppPreflightsRunning:         {states.StateAppPreflightsSucceeded, states.StateAppPreflightsFailed, states.StateAppPreflightsExecutionFailed},
+	states.StateAppPreflightsExecutionFailed: {states.StateAppPreflightsRunning},
+	states.StateAppPreflightsFailed:          {states.StateAppPreflightsRunning, states.StateAppPreflightsFailedBypassed},
+	states.StateAppPreflightsSucceeded:       {states.StateAppPreflightsRunning},
+	states.StateAppPreflightsFailedBypassed:  {states.StateAppPreflightsRunning},
```

Comment:
if i bypass failed preflights i then need to be able to proceed. i should not have to re-run

Reply:
discussed irl. this will happen in a follow up pr